test/app/model/ChessEngine.spec.ts

```diff
@@ -0,0 +1,38 @@
+import { ChessEngine } from '../../../src/app/model/ChessEngine';
+import { PieceWasCaptured, PieceWasMoved, Side, Square, SquareWithPiece } from '../../../src/app/model/Types';
+import { ChessBoard } from '../../../src/app/model/ChessBoard';
+import { Pawn } from '../../../src/app/model/Pawn';
+import { Player } from '../../../src/app/model/Player';
+import 'jest-extended';
+import { Queen } from '../../../src/app/model/Queen';
+
+describe('Chess Engine', () => {
+  it('Given white piece on A2 and black piece on A4, should move white piece to A3 and return expected list of events', () => {
```

Comment:
`and return expected list of events` to raczej nie potrzebne :) Opisujesz wlasnie w tescie moze nawet lepiej tak:

`given white piece on A2 and black piece on A4, when move white piece from A2 to A3, then white piece was moved from A2 to A3`

Co Ty na to?

Reply:
Twój opis brzmi bardziej opisowo :D tak zrobię